app/src/main/java/com/lagradost/cloudstream3/plugins/PluginManager.kt

```diff
@@ -265,16 +266,96 @@ object PluginManager {
         }
 
         main {
-            createNotification(activity, updatedPlugins)
+            createNotification(activity, updatedPlugins, R.string.plugins_updated)
         }
 
-       // ioSafe {
+        // ioSafe {
             afterPluginsLoadedEvent.invoke(true)
-       // }
+        // }
 
         Log.i(TAG, "Plugin update done!")
     }
 
+    /**
+     * Automatically download plugins not yet existing on local
+     * 1. Gets all online data from online plugins repo
+     * 2. Fetch all not downloaded plugins
+     * 3. Download them and reload plugins
+     **/
+    fun downloadNotExistingPluginsAndLoad(activity: Activity) {
+        val newDownloadPlugins = mutableListOf<String>()
+        val urls = (getKey<Array<RepositoryData>>(REPOSITORIES_KEY)
+            ?: emptyArray()) + PREBUILT_REPOSITORIES
+        val onlinePlugins = urls.toList().apmap {
+            getRepoPlugins(it.url)?.toList() ?: emptyList()
+        }.flatten().distinctBy { it.second.url }
+
+        val providerLang = activity.getApiProviderLangSettings()
+        //Log.i(TAG, "providerLang => ${providerLang.toJson()}")
+
+        // Iterate online repos and returns not downloaded plugins
+        val notDownloadedPlugins = onlinePlugins.mapNotNull outer@{ onlineData ->
```

Comment:
not sure why you have to label this

Reply:
You meant the ``outer`` loop? My prev implementation required 2 ``.mapNotNull`` but now I've changed it to only use 1. It can be removed.